### PR TITLE
Modify rule S6504: Clarify intention of rule (APPSEC-1350)

### PR DESCRIPTION
**Pull request by daniel-teuchert-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S6504/docker/rule.adoc [hunk 1]
```diff
@@ -1,30 +1,26 @@
-Ownership of an executable has been assigned to a user other than root. More
-often than not, resource owners have write permissions and thus can edit the
-resource.
+Ownership or write permissions for a file or directory copied to the docker image have been assigned to a user other than root.
```
Comment (egon-okerman-sonarsource) [inline rules/S6504/docker/rule.adoc:1]:
```suggestion
Ownership or write permissions for a file or directory copied to the Docker image have been assigned to a user other than root.
```

File: rules/S6504/docker/rule.adoc [hunk 2]
```diff
@@ -1,30 +1,26 @@
...
 
 Write permissions enable malicious actors, who got a foothold on the container,
```
Comment (egon-okerman-sonarsource) [inline rules/S6504/docker/rule.adoc:3]:
```suggestion
Write permissions enable malicious actors, who have a foothold on the container,
```

File: rules/S6504/docker/rule.adoc [hunk 3]
```diff
@@ -1,30 +1,26 @@
...
 
-Resource ownership is not required; executables can be assigned execute
-permissions using `chmod` if needed.
+If a user is given ownership on a file but no write permissions, the user can still modify it by using his ownership to change the file permissions first. This is why both ownership and write permissions should be avoided.
 
 
 == Ask Yourself Whether
 
-* A non-root user has write permissions for the executable.
+* A non-root user has write permissions or ownership for the resource.
```
Comment (egon-okerman-sonarsource) [inline rules/S6504/docker/rule.adoc:16]: "has (...) ownership for the resource" sounds a bit strange

File: rules/S6504/metadata.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-    "title": "Having executables not owned by root is security-sensitive",
+    "title": "Resources copied to an image should not be modifiable by non-root users",
```
Comment (egon-okerman-sonarsource) [inline rules/S6504/metadata.json:2]: "X should Y" is typically used as a title in vulnerabilities, but this is still a security hotspot.

**Review by egon-okerman-sonarsource — CHANGES_REQUESTED** [review]
Most changes are fine, but it should be decided if this rule should be a vulnerability or a security hotspot before this is merged.

**Comment by daniel-teuchert-sonarsource** [comment]
> Most changes are fine, but it should be decided if this rule should be a vulnerability or a security hotspot before this is merged.

I would still keep this rule as a security hotspot.

**Review by egon-okerman-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**